src/services/urlUtils.ts

```diff
@@ -122,41 +122,59 @@ namespace UrlUtils {
     });
   };
 
-  export const getNavigations = (request: UrlBeautifier.SearchUrlState) => {
+  // tslint:disable-next-line max-line-length
+  export const getNavigations = (state: Store.Indexed<Store.Navigation> | Store.AvailableNavigations, request: UrlBeautifier.SearchUrlState) => {
     return {
-      allIds: getAllIds(request),
-      byId: getById(request),
+      allIds: getAllIds(state, request),
+      byId: getById(state, request),
     };
   };
 
-  export const getAllIds = (request: UrlBeautifier.SearchUrlState) => {
-    return request.refinements.reduce((fields, refinement) => {
-      if (!fields.includes(refinement.field)) {
-        fields.push(refinement.field);
+  // tslint:disable-next-line max-line-length
+  export const getAllIds = (state: Store.Indexed<Store.Navigation> | Store.AvailableNavigations, request: UrlBeautifier.SearchUrlState) => {
+    const allIds = state.allIds;
+    request.refinements.forEach((refinement) => {
```

Comment:
You can `filter` first, then spread the resulting array to `push` if you think that might be clearer.